sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetTablesOperation.scala

```diff
@@ -72,29 +74,49 @@ private[hive] class SparkGetTablesOperation(
       authorizeMetaGets(HiveOperationType.GET_TABLES, privObjs, cmdStr)
     }
 
-    val tablePattern = convertIdentifierPattern(tableName, true)
+    // Tables and views
     matchingDbs.foreach { dbName =>
       catalog.getTablesByName(catalog.listTables(dbName, tablePattern)).foreach { catalogTable =>
```

Comment:
in the proposed api, this would call `catalog.listTables(dbName, tablePattern, false)` as local temp views are handled elswhere.